stan/math/prim/scal/fun/as_scalar.hpp

```diff
@@ -0,0 +1,18 @@
+#ifndef STAN_MATH_PRIM_SCAL_FUN_AS_SCALAR_HPP
+#define STAN_MATH_PRIM_SCAL_FUN_AS_SCALAR_HPP
+
+namespace stan {
+namespace math {
+
+/**
+ * Converts input to a scalar. For scalar arguments this is an identity
```

Comment:
This isn't quite true.  For scalar argument that are integers, they get converted to double values.  If that's not the intent, there needs to be another specialization or this just needs to be replaced with

```
<template typename T> 
inline const T& as_scalar(const T& x) { return x; }
```